ISISComputingGroup/ibex_system_tests
# Ticket1436 push to git on clean

This is for ticket ISISComputingGroup/IBEX#1436

The cleanup script in the system tests now stops the blockserver BEFORE deleting rcptt files from the config, then the blockserver is restarted.
Note that when the blockserver is first launched, procServ is set to autorestart it if it stops. So to stop it properly we need to toggle the procServ autorestart flag to False (this would be done manually with CTRL+T in the console). The blockserver then restarts simply by re-toggling the flag to True.

I have done a bit of refactoring of the current code too.

To test:
Run the instrument and run a couple of system tests that create rcptt files and check that they pass. In particular, run the same system test consecutively more than once, to verify that any file created is correctly deleted without anything complaining.

Note that I have added some extra checks in the cleanup script to verify that the console process has opened and worked correctly: to do this I am looking at the output and errors from the communicate() method of the python subprocess (I have looked into the docs of python subprocess, couldn't find a better way, but if you know a better one I can change it!). The error scenarios I have tried are:
- instrument is not running --> error for connection refused
- instrument running but host name wrong in script --> error for host lookup
- instrument running but console name wrong --> subprocess output says console not found.

Without these checks, the python subprocess doesn't throw any error, and the script would continue running happily! Now instead it stops and logs the error message (note if the script stops, its output doesn't make it to the console.log generated by rcptt, but you can see it in the cleanIBEXServer.log file)
(if you do modify the script to try these scenarios, don't forget to update the script in the CleanIBEXServerFile rcptt context, as described in the wiki)
